lib/libzfsacl/zfsacltests/test_nfsv4acl.py

```diff
@@ -0,0 +1,1531 @@
+        entry0 = acl.get_entry(0)
+        self.assertEqual(0, entry0.idx, "Default ACE 0 idx is not 0")
+        self.assertEqual(libzfsacl.ENTRY_TYPE_ALLOW, entry0.entry_type,
+                         "Default ACE 0 is not ENTRY_TYPE_ALLOW")
+        self.assertEqual(0, entry0.flagset,
+                         "Default ACE 0 flagset is not NO_INHERIT")
+        self.assertEqual(libzfsacl.WHOTYPE_USER_OBJ, entry0.who[0],
+                         "ACE 0 who type is not USER_OBJ")
+        self.assertEqual(self.USER_OBJ_PERMSET,
+                         entry0.permset & self.USER_OBJ_PERMSET,
+                         "Default ACE 0 permset does not match"
+                         "USER_OBJ_PERMSET")
+
+    # Test default GROUP_OBJ ACE is present
+    def test_007_default_ace_group_obj(self):
+        acl = libzfsacl.Acl(path=self.MOUNTPT)
+        entry1 = acl.get_entry(1)
+        self.assertEqual(1, entry1.idx, "Default ACE 1 idx is not 1")
+        self.assertEqual(libzfsacl.ENTRY_TYPE_ALLOW, entry1.entry_type,
+                         "Default ACE 1 is not ENTRY_TYPE_ALLOW")
+        self.assertEqual(libzfsacl.WHOTYPE_GROUP_OBJ, entry1.who[0],
+                         "ACE 1 who type is not GROUP_OBJ")
+
+    # Test default EVERYONE ACE is present
+    def test_008_default_ace_everyone(self):
+        acl = libzfsacl.Acl(path=self.MOUNTPT)
+        entry2 = acl.get_entry(2)
+        self.assertEqual(2, entry2.idx, "Default ACE 2 idx is not 1")
+        self.assertEqual(libzfsacl.ENTRY_TYPE_ALLOW, entry2.entry_type,
+                         "Default ACE 2 is not ENTRY_TYPE_ALLOW")
+        self.assertEqual(0, entry2.flagset,
+                         "Default ACE 2 flagset is not NO_INHERIT")
+        self.assertEqual(libzfsacl.WHOTYPE_EVERYONE, entry2.who[0],
+                         "ACE 2 who type is not EVERYONE")
+
+    # Test an ACE can be appended
+    def test_009_append_an_ace(self):
+        os.makedirs(self.TDIR)
+        dacl = libzfsacl.Acl(path=self.TDIR)
+        orig_cnt = dacl.ace_count
+        newEntry = dacl.create_entry()
+        newEntry.entry_type = libzfsacl.ENTRY_TYPE_ALLOW
+        newEntry.who = (libzfsacl.WHOTYPE_USER, self.ZFS_ACL_STAFF1_UID)
+        dacl.setacl(path=self.TDIR)
+        new_cnt = libzfsacl.Acl(path=self.TDIR).ace_count
+        os.rmdir(self.TDIR)
+        self.assertEqual(orig_cnt + 1, new_cnt, "Failed to add an ace")
+
+    # Test an ACE can be prepended
+    def test_010_prepend_an_ace(self):
+        os.makedirs(self.TDIR)
+        dacl = libzfsacl.Acl(path=self.TDIR)
+        orig_cnt = dacl.ace_count
+        newEntry = dacl.create_entry(0)
+        newEntry.entry_type = libzfsacl.ENTRY_TYPE_ALLOW
+        newEntry.who = (libzfsacl.WHOTYPE_USER, self.ZFS_ACL_STAFF1_UID)
+        dacl.setacl(path=self.TDIR)
+        new_cnt = libzfsacl.Acl(path=self.TDIR).ace_count
+        os.rmdir(self.TDIR)
+        self.assertEqual(orig_cnt + 1, new_cnt, "Failed to add an ace")
```

Comment:
Instead of checking the count, I guess we can check if new ACE is at the index 0.

Reply:
Thanks, I have updated.